### DISCUSSION TITLE
How to use multiple Cloud image name using next-cloudnary package.

### DISCUSSION THREAD

**Discussion by kashyapshiv** [cloudinary-community/next-cloudinary#215] [Q&A]
Issue Statement:
I was using free plan of cloudnary and I had different cloud name and generating image with them. Now I have bought basic plan and got my own cloud name.

The issue is when I am setting new cloud name in my .env file, old images generated with older cloud name are not rendering and same if I keep old one, new images generating won't be rendered.

Configuration : 
Technology/Framework : NextJS (13.3.4)
Cloudnary package: next-cloudinary (4.4.0)

Using CldImage component.
import { CldImage } from 'next-cloudinary';

`<CldImage
              src={searchPromptOptions?.image}
              alt="uploaded image"
              width={400}
              height={400}
             
            />`

**Comment by kashyapshiv** [comment]
@colbyfayock please help with the issue, how can I resolve it.

**Comment by colbyfayock** [comment]
hey @kashyapshiv to be clear - did you upgrade the free account to paid or did you create a new account, separate from your original free account?

images wouldn't automatically transfer over to a new account, you would need to upload the images from your original account to your new account

if thats the case, is there a reason you didnt upgrade the original free account?

using the `.env` you're only able to configure one cloud account at a time, it's the global configuration for the component

if you'd like to have an app that uses multiple clouds in one app, you can configure individual components with the cloud name:

```
<CldImage
  ...
  config={{
    cloud: {
      cloudName: '<Your Cloud Name>'
    }
  }}
/>
```

This could be a little tedious to do on every component so you could wrap that with a separate component to use interchangeably throughout the app

**Reply by kashyapshiv** [reply]
Hey @colbyfayock , thanks for answering. I didn't switch the account, I just upgraded from free plan to basic plan and set my custom cloud name. So can I upload my old images with older cloud name in the links with newer one?

Meanwhile I will definitely  try with the solution given above.

**Reply by colbyfayock** [reply]
i dont think the solutions relevant then, you should be able to update your cloud name in the .env file and if your images still exist inside of your media library (which they should), it should work fine

do you have an example image using both the old and new cloud names that you can share the URL for?

**Reply by kashyapro** [reply]
Thanks @colbyfayock , the above solution is working for me, the config one. 

This URL works with older cloud name but not with new one https://res.cloudinary.com/dfpgenl5l/image/upload/c_limit,w_600/f_auto/q_auto/v1685526194/etjqwkpuqdqmijrnujc0?_a=BAVAUYBx0

And this works with new one:
https://res.cloudinary.com/qolaba/image/upload/c_limit,w_600/f_auto/q_auto/v1687436670/uxfzlwx4kqqllj67mkd6?_a=BAVAS0Bx0

**Reply by colbyfayock** [reply]
it's showing a 404 meaning that it can't find that asset, so either its not available in the Media Library the reference to the image ID is wrong. you can try searching for the image ID in the library to see if its there
